libr/util/strbuf.c

```diff
@@ -293,29 +319,39 @@ R_API ut8 *r_strbuf_getbin(RStrBuf *sb, int *len) {
 	return (ut8 *)(sb->ptr ? sb->ptr : sb->buf);
 }
 
+static inline char *drain(RStrBuf *sb) {
+	return sb->ptr
+		? sb->weakref
+			? r_mem_dup (sb->ptr, sb->ptrlen)
+			: sb->ptr
+		: strdup (sb->buf);
+}
+
 R_API char *r_strbuf_drain(RStrBuf *sb) {
 	r_return_val_if_fail (sb, NULL);
-	char *ret = sb->ptr ? sb->ptr : strdup (sb->buf);
+	char *ret = drain (sb);
```

Comment:
can't we just return sb->ptr in case of weakref?

Reply:
No, drain should always give you a heap-allocated string that you own after the return. At least that is my understanding of it. Most of the time you don't want to drain a weak strbuf, but I think it makes more sense to always pass the ownership to the caller instead of only in some cases.